### PR TITLE
Move scripts used by all-core.sh to the framework

### LINKED ISSUE #74: Move scripts used by `all-core.sh` to the framework
Move the following scripts to the framework:
```
scripts/output_env.sh
tests/scripts/quiet/*
```

Those are the scripts that are used by `all-core.sh`. As of writing they're identical between 3.6 and development, this task should be resolved by just moving them and adapting paths were needed.

### PR DESCRIPTION
**Pull request by eleuzi01**
Resolves #74 

Development PR: https://github.com/Mbed-TLS/mbedtls/pull/9788
3.6 PR: https://github.com/Mbed-TLS/mbedtls/pull/9789

### CODE REVIEW

**Comment by eleuzi01** [comment]
When I did #56, I moved files from tests/scripts to framework/scripts. I don't really remember what was the reason behind that but I followed the same logic here. Could the reviewers let me know whether the files from tests/scripts should be under framework/test/scripts or just framework/scripts and I'll fix that if needed, thanks!

File: scripts/all-core.sh [hunk 1]
```diff
@@ -861,7 +861,7 @@ pre_check_tools () {
         *) set "$@" RUN_ARMCC=0;;
     esac
     # Use a path relative to the currently-sourced file.
-    "$@" "${BASH_SOURCE%/*}"/../../scripts/output_env.sh
+    "$@" "${BASH_SOURCE%/*}"/../../framework/scripts/output_env.sh
```
Comment (mpg) [inline scripts/all-core.sh:864]: This is correct but I think we can simplify this by removing `../../framework/scripts` now that the two files live in the same directory: `"${BASH_SOURCE%/*}"/output_env.sh` should do the trick.

**Review by mpg — DISMISSED** [review]
LGTM. Just suggesting an optional minor improvement (I think/hope).

**Comment by mpg** [comment]
The failures in the CI are expected and are fixed in the companion PRs which both have a green CI, so everything's OK.

We'll just need to merge the companion PRs quickly after merging this one.

**Comment by eleuzi01** [comment]
> When I did #56, I moved files from tests/scripts to framework/scripts. I don't really remember what was the reason behind that but I followed the same logic here. Could the reviewers let me know whether the files from tests/scripts should be under framework/test/scripts or just framework/scripts and I'll fix that if needed, thanks!

@mpg this question got lost above, wdyt?

**Review by valeriosetti — DISMISSED** [review]

**Comment by mpg** [comment]
> When I did #56, I moved files from tests/scripts to framework/scripts. I don't really remember what was the reason behind that but I followed the same logic here. Could the reviewers let me know whether the files from tests/scripts should be under framework/test/scripts or just framework/scripts and I'll fix that if needed, thanks!
> @mpg this question got lost above, wdyt?

Ah indeed I had completely missed this question that ended up hidden in the long list of commits, thanks for raising it again!

To be honest I'm not sure, when I moved `all-core.sh` I did the same, mostly following what seemed to be the pattern from existing files. (Which is usually a good guess, but of course explicitly raising the question as you did here is better.)

I'm _assuming_ the reason is that the distinction between `scripts` and `tests/scripts` was never super useful in the first place, and makes things more complicated when it comes to importing python modules in one directory from modules/scripts in the other (currently we have a tool for doing it one way but not the other), so we decided to drop it while we're moving things.

Also, I guess since we're adding a new `framework` component to the path it's nice to remove the `tests` component at the same time to avoid making the tree too deep (though that's a relatively weak reason).

@ronald-cron-arm @gilles-peskine-arm Any thoughts you'd like to share?

**Review by mpg — APPROVED** [review]
LGTM, thanks!

**Review by valeriosetti — APPROVED** [review]

**Review by valeriosetti — APPROVED** [review]

**Comment by mpg** [comment]
CI is green in both consuming PRs - which both point to the tip of this branch.

**Pull request merged**